### DISCUSSION TITLE
Framework ignoring data loaders

### DISCUSSION THREAD

**Discussion by jamesking** [observablehq/framework#1679] [Q&A] [answered]
Framework seems to be ignoring a data loaders in my project. 

I have 2 data loaders: `./src/data/feed.xml.js` and `./src/data/launches.csv.js` (from the Framework examples). 

![CleanShot 2024-09-24 at 17 43 09@2x](https://github.com/user-attachments/assets/f1a5172b-b88a-40f1-9cf2-c6d5bffccfef)

After `npm run build` the `./dist/_file` directory is pretty bare. The only file that appears in the  directory is the Observable favicon.

![CleanShot 2024-09-24 at 17 43 53@2x](https://github.com/user-attachments/assets/d4c25f7f-edc8-41cb-b832-7626ebc2f308)

Any help troubleshooting appreciated.

**Comment by Fil** [comment, marked as answer]
Any file **must** be referenced statically (say, in a page) if you want it to be built. In other words, data loaders are only activated if they are used.

(You can reference a file without loading it, by using a `link[rel]` or a bare `FileAttachment("feed.xml")` with no loading method.)

**Reply by jamesking** [reply]
Thanks! I was totally stumped.
